### DISCUSSION TITLE
Questions about uploading a new dataset (to the HF Datasets hub)

### DISCUSSION THREAD

**Discussion by rfernand2** [huggingface/datasets#5423] [Q&A] [answered]
Some questions I have not been able to find by searching internet:
   1. How do I define the subsets (tasks) in my dataset?  I would expect to be able to create subdirectories for each task that contain the split files for that task, but subdirectories don't seem to be supported.
   2. How do I define my splits so that the dataset preview tool can put them into the split dropdown?
   3. Is *.jsonl supported for uploading my split files?  I tried using a tab-separated file as *.txt and *.csv, but neither seemed to work (did not have the col name list on the first line)
   4. How does one publish a dataset to the global namespace, like "glue" has been published?
Thanks!

**Comment by rfernand2** [comment, marked as answer]
I found this HF doc file about using a dataset loading script to define multiple subsets/tasks/configs for your dataset and to define the features/columns of each config:  https://huggingface.co/docs/datasets/dataset_script.  It also covers how to define splits and load data.  In addition, it explains that global dataset names (not prefixed with a username) are a legacy from when datasets were hosted in github, and seems to imply that all new datasets must  be created with a username or organization prefix.

**Reply by rfernand2** [reply]
Also, when the docs are unclear or ambiguous, it's good to look at the SuperGlue loading script as a reference of what really works: https://huggingface.co/datasets/super_glue/blob/main/super_glue.py.  What I misunderstood initially is that a dataset loading script was needed for my dataset, even though I was storing my datafiles in the HF datasets hub (for many reasons).  In fact, I think it would be simpler for HF to recommend that all datasets use a loading script and list the features it makes possible (vs. the default of just loading your data files).  This should be mentioned in the tutorial of sharing your data, as well as in the reference section ("Dataset Repository").

**Reply by mariosasko** [reply]
Hi! Indeed, specifying subtasks/configs requires the dataset script approach. If I'm correct, @polinaeterna is working on a no-code approach where one can define subtasks in the README metadata.

cc @stevhliu for the doc suggestions - we prefer a no-code approach when uploading a dataset, but a mention of the dataset script approach still makes sense there IMO. Also, I noticed we should update [this](https://huggingface.co/docs/datasets/upload_dataset#create-a-dataset-card) section regardless - the process is now easier thanks to the Hub's metadata UI update.
